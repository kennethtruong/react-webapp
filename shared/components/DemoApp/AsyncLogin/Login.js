import React, { Component, PropTypes } from 'react';
import Helmet from 'react-helmet';
import { observer, inject } from 'mobx-react';
import Redirect from 'react-router-dom/Redirect';

class Login extends Component {
  static propTypes = {
    auth: PropTypes.object,     // eslint-disable-line
    location: PropTypes.object, // eslint-disable-line
  };

  state = {
    redirectToReferrer: false,
  }

  onLogin = async () => {
    const auth = this.props.auth;
    await auth.login('test', 'test');
    this.setState({ redirectToReferrer: true });
  }

  render() {
    const { from } = this.props.location.state || { from: { pathname: '/' } };
    const { redirectToReferrer } = this.state;

    if (redirectToReferrer) {
      return (
        <Redirect to={from} />
      );
    }

    return (
      <article>
        <Helmet title="Counter" />
        <button onClick={this.onLogin}>Login</button>
      </article>
    );
  }
}

export default inject('auth')(observer(Login));
