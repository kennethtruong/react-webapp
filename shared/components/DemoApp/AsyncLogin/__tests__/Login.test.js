/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';

import Login from '../Login';
import getAuth from '../../../../store/Auth';

describe('<Login />', () => {
  test('renders', () => {
    const wrapper = shallow(<Login auth={getAuth()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
