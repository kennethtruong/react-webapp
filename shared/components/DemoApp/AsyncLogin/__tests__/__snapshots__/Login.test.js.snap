exports[`<Login /> renders 1`] = `
<Login
  auth={
    Auth {
      "errorMessage": undefined,
      "user": undefined,
    }
  } />
`;
